nsxt/resource_nsxt_vpc_service_profile.go

```diff
@@ -0,0 +1,445 @@
+			SchemaType:   "string",
+			SdkFieldName: "SecurityProfile",
+		},
+	},
+	"qos_profile": {
+		Schema: schema.Schema{
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "string",
+			SdkFieldName: "QosProfile",
+		},
+	},
+	"dns_forwarder_config": {
+		Schema: schema.Schema{
+			Type:     schema.TypeList,
+			MaxItems: 1,
+			Elem: &metadata.ExtendedResource{
+				Schema: map[string]*metadata.ExtendedSchema{
+					"cache_size": {
+						Schema: schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  1024,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "int",
+							SdkFieldName: "CacheSize",
+						},
+					},
+					"log_level": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validation.StringInSlice(vpcServiceProfileLogLevelValues, false),
+							Optional:     true,
+							Default:      model.PolicyVpcDnsForwarder_LOG_LEVEL_INFO,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "string",
+							SdkFieldName: "LogLevel",
+						},
+					},
+					"conditional_forwarder_zone_paths": {
+						Schema: schema.Schema{
+							Type: schema.TypeList,
+							Elem: &metadata.ExtendedSchema{
+								Schema: schema.Schema{
+									Type:         schema.TypeString,
+									ValidateFunc: validatePolicyPath(),
+								},
+								Metadata: metadata.Metadata{
+									SchemaType: "string",
+								},
+							},
+							Optional: true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "list",
+							SdkFieldName: "ConditionalForwarderZonePaths",
+						},
+					},
+					"default_forwarder_zone_path": {
+						Schema: schema.Schema{
+							Type:         schema.TypeString,
+							ValidateFunc: validatePolicyPath(),
+							Optional:     true,
+						},
+						Metadata: metadata.Metadata{
+							SchemaType:   "string",
+							SdkFieldName: "DefaultForwarderZonePath",
+						},
+					},
+				},
+			},
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "struct",
+			SdkFieldName: "DnsForwarderConfig",
+			ReflectType:  reflect.TypeOf(model.PolicyVpcDnsForwarder{}),
+		},
+	},
+}
+
+func resourceNsxtVpcServiceProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtVpcServiceProfileCreate,
+		Read:   resourceNsxtVpcServiceProfileRead,
+		Update: resourceNsxtVpcServiceProfileUpdate,
+		Delete: resourceNsxtVpcServiceProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+		Schema: metadata.GetSchemaFromExtendedSchema(vpcServiceProfileSchema),
+	}
+}
+
+func resourceNsxtVpcServiceProfileExists(sessionContext utl.SessionContext, id string, connector client.Connector) (bool, error) {
+	var err error
+	parents := getVpcParentsFromContext(sessionContext)
+	client := clientLayer.NewVpcServiceProfilesClient(connector)
+	_, err = client.Get(parents[0], parents[1], id)
```

Comment:
It is simpler to take org/project/VPC from context than loading these into parents array and using from there

Reply:
For the generator this way is a cleaner and more readable option I think. The only difference is function name, rather than rewriting the whole block.